### PR TITLE
`General`: Use short result string as default

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

The result string can take up much space, especially for programming exercises. Which is the reason for a short form (only achieved points).

### Description

This short form is now used as the default for results. Only the tutor assessment for programming exercises directly shows the passed tests to help the tutor quickly see this information when assessing or replying to a complaint.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Tutor
- 1 Programming Exercise with some submissions

1. Log in to Artemis
2. Check that the shot result string is used in almost all places (e.g. scores table, student view, ...)
3. Check that the tutor sees the relevant information quickly

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Current long format (short format would omit tests and issues here):
Bildschirmfoto 2023-04-27 um 15 21 50

### CODE REVIEW

**Review by danielbode — APPROVED** [review]
Code LGTM

**Review by pal03377 — APPROVED** [review]
I'm not 100% sure about which parts of the application should now show which version. I'll just list the places I found while testing here and where I was unsure. If they are ok, it's fine for me 👍  Seeing the shorter results is not what I'm used to, but it looks nice.

Exercise Management Page: Long. Should it be this way?
screenshot-2023-04-29_001291

Assessment Dashboard: Short. Maybe the tutors want to see the long version here?
screenshot-2023-04-29_001292


Optional suggestion: Also I quickly looked through the related code and I think we could change the type of the parameter `short` from `boolean | undefined` to just `boolean` in a few places, e.g. for the function `getResultStringDefinedParameters`. 🤔 What do you think?

Tl;dr: Code obviously looks good, functionality seems to be as expected (TS1)

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Review by pal03377 — APPROVED** [review]
re-approve because only tests were changed

**Review by danielbode — APPROVED** [review]
Reapprove code

**Review by TheZoker — APPROVED** [review]
Changes look good to me! 👍

**Review by tobias-lippert — APPROVED** [review]
Code reapprove

**Review by bassner — APPROVED** [review]
